ScoutSuite/core/processingengine.py

```diff
@@ -40,38 +37,30 @@ def run(self, cloud_provider, skip_dashboard=False):
+                        rule_dict[attr] = getattr(rule, attr)
                 try:
                     setattr(rule, 'checked_items', 0)
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['items'] = recurse(
-                        cloud_provider.services, cloud_provider.services, path, [], rule, True)
+                    for a in dir(rule):
+                        if not a.startswith("__") and a not in [
+                            "set_definition",
+                            "get_attribute",
+                            "to_string",
+                        ]:
+                            rule_dict[str(a)] = getattr(rule, a)
                     if skip_dashboard:
                         continue
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['dashboard_name'] = \
-                        rule.dashboard_name
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['checked_items'] = \
-                        rule.checked_items
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['flagged_items'] = \
-                        len(cloud_provider.services[service][self.ruleset.rule_type][rule.key]['items'])
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['service'] = rule.service
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['rationale'] = \
-                        rule.rationale if hasattr(rule, 'rationale') else None
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['remediation'] = \
-                        rule.remediation if hasattr(rule, 'remediation') else None
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['compliance'] = \
-                        rule.compliance if hasattr(rule, 'compliance') else None
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['references'] = \
-                        rule.references if hasattr(rule, 'references') else None
+                    rule_dict['flagged_items'] = len(cloud_provider.services[service][self.ruleset.rule_type][rule.key]['items'])
                 except Exception as e:
                     print_exception('Failed to process rule defined in %s: %s' % (rule.filename, e))
                     # Fallback if process rule failed to ensure report creation and data dump still happen
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['checked_items'] = 0
-                    cloud_provider.services[service][self.ruleset.rule_type][rule.key]['flagged_items'] = 0
+                    rule_dict['checked_items'] = 0
+                    rule_dict['flagged_items'] = 0
+
+                cloud_provider.services[service][self.ruleset.rule_type][rule.key] = rule_dict
```

Comment:
I **think** that with this assignment you would be potentially losing the info stored at cloud_provider.services[service][self.ruleset.rule_type][rule.key]['items'], unless this is copied to rule_dict somewhere I have not seen

Reply:
Ah, yeah, you're probably right.  I wanted this to be an `UPSERT` (for lack of a better term) of the existing dictionary.  I know that I am running this and getting some inconsistent results so this might be why...